pcdsutils/ext_scripts.py

```diff
@@ -27,28 +50,28 @@ def call_script(args, timeout=None, ignore_return_code=False):
         raise
 
 
-cache = {}
-
-
-def cache_script(args, timeout=None, ignore_return_code=False):
-    key = ' '.join(args)
-    try:
-        return cache[key]
-    except KeyError:
-        output = call_script(args, timeout=timeout,
-                             ignore_return_code=ignore_return_code)
-        cache[key] = output
-        return output
+@functools.lru_cache(maxsize=None)
+def get_hutch_name(timeout=10):
```

Comment:
Only unfortunate part about `lru_cache` is that it depends on the arguments - so a `timeout=10` will be cached separately from `timeout=None` and so on.

To that end, do we really need a customizable timeout for specific functions like this?

Reply:
Oh no

Reply:
In this case nobody actually changes the timeout ever... and the previous implementation had the same bug. But this still worries me out of principle.

Reply:
I think there's an easy way to fix this

Reply:
My "unfortunate" above was a poor choice - it makes a lot of sense from the perspective of the cache decorator. Hutch-specific caching in `get_ami_proxy` would fail without this, after all.

Recently my go-to solution in similar situations to this has been adding a context manager. Something like:

```
with script_timeout_context(10):
   run_script(... no_timeout_here... )
```

If we can live without the `timeout=` API, the above could be an option. If not, roll-back I suppose?

Reply:
I'm going to roll-back and save this question for another day